service.go

```diff
@@ -156,6 +161,37 @@ func (s *Service) start() error {
 		}),
 		nats.MaxReconnects(-1), // never give up, the default was 60
 	}
+
+	if s.conf.Username != "" {
+		connectOpts = append(connectOpts, nats.UserInfo(s.conf.Username, s.conf.Password))
+	}
+
+	if s.conf.Token != "" {
+		connectOpts = append(connectOpts, nats.Token(s.conf.Token))
+	}
+
+	// Enable TLS if configured
+	if s.conf.TLS.HasCA() || s.conf.TLS.HasCertWithKey() {
+		ctx := s.opt.Context
+		if ctx == nil {
+			ctx = context.Background()
+		}
+		c, cancel := context.WithCancel(ctx)
+		defer cancel()
```

Comment:
Bug: this will ALWAYS cancel the manager. You only want to do it before `return err` for `TLSConfig()`